### ISSUE TITLE
package.version triggers a mismatch when using workspace:* protocol

### ISSUE THREAD

**Issue by battis** [JamieMason/syncpack#238]
## Description

In a monorepo using `pnpm`, each package has its own `package.json` file that lists its current version in the `version` field and depends on another package using the `workspace:*` protocol:

`/packages/a/package.json`
```json
{
  "name": "@app/a",
  "version": "1.2.3"
}
```

`/packages/b/package.json`
```json
{
  "name": "@app/b",
  "dependencies": {
    "@app/a": "workspace:*"
  }
}
```

With the following modified [example](https://jamiemason.github.io/syncpack/examples/pnpm-workspace-protocol/) config (modified to include non-dev dependencies as well):

`/.syncpackrc`
```json
{
  "versionGroups": [
    {
      "label": "Use workspace protocol when developing local packages",
      "dependencies": ["$LOCAL"],
      "pinVersion": "workspace:*"
    }
  ]
}
```

When running `syncpack` this is identified as a mismatch:

```sh
syncpack list

= Use workspace protocol when developing local packages ========================
     1x @app/a:
        ✘ 1.2.3 → workspace:* [PinnedMismatch]
```

Even more excitingly, if `sync fix-mismatches` is run, the the `@app/a` `package.json` is updated thus:

`/packages/a/package.json`
```json
{
  "name": "@app/a",
  "version": "workspace:*"
}
```

## Suggested Solution

If I am correctly understanding what is going on, then the `package.version` field should not be identified as a mismatch -- only the dependency versions should be. That is, it should be fine for local packages to maintain their own versioning within the monorepo, but the _other_ packages should not be depending on a specific version (per `.syncpackrc`) but instead depend only on `workspace:*.

And `fix-mismatches` should _not_ update the `version` field of a package, since that is an information source, not a dependency.

## Help Needed

It is entirely possible that I'm misunderstanding what `syncpack` is reporting, but I _think_ this is a bug and not desired behavior?

**Comment by JamieMason** [comment]
Thanks @battis, this catches people out all the time.

Try this?

```diff
{
  "versionGroups": [
    {
      "label": "Use workspace protocol when developing local packages",
      "dependencies": ["$LOCAL"],
+     "dependencyTypes": ["!$LOCAL"],
      "pinVersion": "workspace:*"
    }
  ]
}
```

**Comment by JamieMason** [comment]
Great, this is what the config is trying to do:

[`local`](https://jamiemason.github.io/syncpack/guide/local-package-versions/#the-local-dependency-type) is a keyword just like [`dev`, `prod`, `peer` etc](https://jamiemason.github.io/syncpack/config/dependency-types/#default-dependency-types) which refers to the `version` properties of the package.json files from your own packages being developed in your monorepo. 

Instead of writing out all the dependency names we use these keywords, a bit like variables.

When you use `!local` instead of `!local` you're saying: include all the other types except this one.

**Comment by battis** [comment]
Elegant!

**Comment by mikededo** [comment]
Hmm, even though I'm using a name [specified here](https://jamiemason.github.io/syncpack/config/syncpackrc/#configuration-file) at the root level, somehow, unless I use the `--config <path>` flag, the configuration is ignored 🤔

**Comment by JamieMason** [comment]
I wonder if it is similar to [this problem](https://github.com/JamieMason/syncpack/issues/229#issuecomment-2205646499), where an error is happening in the config but is not being surfaced yet. 

`SYNCPACK_VERBOSE=true syncpack list` may throw up clues, but try renaming the config file to .cjs

**Comment by mikededo** [comment]
Well, working with `.syncpackrc.json` seems to work fine, so that should resolve the issue 🙏🏼👍🏼

**Comment by JamieMason** [comment]
Great, my best guess is maybe there was some ESM issue going on that syncpack is swallowing (will be fixed in #229), but I'm not sure.
